0008-interledger-quoting-protocol/0008-interledger-quoting-protocol.md

```diff
@@ -1,7 +1,104 @@
+* The **receiver** is the party being credited at the end of the overall chain.
+* The **destination** receives money from the connector in a single link. In a simple payment, the destination is also the receiver.
+* A **connector** facilitates the payment between the source and destination in a single link. In a multiple-hop payment, there are multiple connectors, each of which is the destination of one link and the source of the next.
+
+## Get Quote (HTTP)
+
+This is the only API method in the HTTP+JSON version of the ILQP.
+
+### Request Format
 
-**TODO**: What's the best way to link this?
+```
+GET /
+```
+
+The request uses the following query parameters:
+
+| Field                         | Type        | Description                    |
+|:------------------------------|:------------|:-------------------------------|
+| `source_address`              | ILP Address | Address of the source's account in the source ledger. |
+| `destination_address`         | ILP Address | Address of the destination's account in the destination ledger. |
+| `source_amount`               | Number      | Fixed amount to debit from the source's account. (Required unless `destination_amount` specified.) |
+| `destination_amount`          | Number      | Fixed amount to credit to the destination's account. (Required unless `source_amount` specified.) |
+| `destination_expiry_duration` | Number      | (Optional) Number of milliseconds before the transfer in the destination ledger expires. |
+| `source_expiry_duration`      | Number      | (Optional) Number of milliseconds before the transfer in the source ledger expires. |
+| `slippage`                    | Number      | (Optional) Custom slippage ***amount? percentage?*** to assume for the exchange between ledgers. Lower slippage means the payment is cheaper but more likely to fail due to market movement. |
+
+The request must specify either `source_amount` or `destination_amount` but not both.
+
+Example request:
+
+```
+GET https://connector.example/?source_amount=100.25\
+  &source_address=example.eur-ledger.alice\
+  &destination_address=example.usd-ledger.bob\
+  &source_expiry_duration=6000\
+  &destination_expiry_duration=5
+```
+
+### Response Format
+
+The response uses the HTTP status code 200 OK and contains a JSON object with the following fields:
+
+| Field                         | Type        | Description                    |
+|:------------------------------|:------------|:-------------------------------|
+| `source_connector_account`    | ILP Address | The address of the connector's account in the source ledger where it should receive the first transfer. |
+| `source_ledger`               | ILP Address | The address of the ILP-enabled ledger where the source account is. |
+| `source_amount`               | String      | Decimal number amount of currency the ***TODO: connector's account should receive(?)*** in the source ledger. |
```

Comment:
> I'm concerned that the assumption that ledger fees are paid out of bound doesn't generalize.

Paying ledger fees out of band is a choice, not an assumption. We're saying that it is the responsibility of the sender of a local transfer to pay the fee. Each connector will recoup that cost from the person asking them to do something.

If the fees are high enough that the connector can't risk getting stuck with the cost of failed payments, it can simply collect some fees upfront when the user registers. Note that the connector and its customer/peer share a ledger by definition, so they can pay each other with a simple local transfer.